src/Desktop.Analyzers/UnitTests/MarkAllNonSerializableFieldsTests.cs

```diff
@@ -45,16 +50,122 @@ Imports System
-                    Friend s2 As String;
-                    Private s3 As String;
-                    Public i1 As Integer;
-                    Friend i2 As Integer;
-                    Private i3 As Integer;
+                    Public s1 As String
+                    Friend s2 As String
+                    Private s3 As String
+
+                    Public i1 As Integer
+                    Friend i2 As Integer
+                    Private i3 As Integer
+
+                    Public b1 As Boolean
+                    Friend b2 As Boolean
+                    Private b3 As Boolean
                 End Class");
         }
 
         [Fact]
+        public void CA2235WithConstPrimitiveFields()
+        {
+            VerifyCSharp(@"
+                using System;
+    
+                [Serializable]
+                public class CA2235WithConstPrimitiveFields
+                {
+                    public const int i1 = 42;
+                    internal const int i2 = 54;
+                    private const int i3 = 96;
+                }");
+
+            VerifyBasic(@"
+                Imports System
+
+                <Serializable>
+                Public Class CA2235WithConstPrimitiveFields 
+
+                    Public Const i1 As Integer = 42
+                    Friend Const i2 As Integer = 54
+                    Private Const i3 As Integer = 96
+                End Class");
+        }
+
+        [Fact]
+        public void CA2235WithPrimitiveGetOnlyProperties()
+        {
+            VerifyCSharp(@"
+                using System;
+    
+                [Serializable]
+                public class CA2235WithPrimitiveGetOnlyProperties
+                {
+                    public int I1 { get; } = 42;
+                    internal int I2 { get; } = 54;
+                    private int I3 { get; } = 96;
+                }");
+
+            VerifyBasic(@"
+                Imports System
+
+                <Serializable>
+                Public Class CA2235WithPrimitiveGetOnlyProperties 
+
+                    Public Property I1 As Integer
+                        Get
+                            Return 42
+                        End Get
+                    End Property
+
+                    Friend Property I2 As Integer
+                        Get
+                            Return 54
+                        End Get
+                    End Property
+
+                    Private Const i3 As Integer
+                        Get
+                            Return 96
+                        End Get
+                    End Property
+
+                   ' Using auto-implemented property syntax
+                    Public Property I1 As Integer = 42
+                    Friend Property I2 As Integer = 54
+                    Private Const i3 As Integer = 96
+                End Class");
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/3898")]
+        public void CA2235WithSerializableLibraryTypes()
```

Comment:
Add tests for serializable library types. This is a real bug, reported by @bkoelman and other, and so this test fails and is skipped for now, until the bug is fixed.

The root cause is that `Serializable` is not a "real" attribute. In the compiled assembly, it is represented as a bit in the metadata representing the type, not as a normal custom attribute, and so it is not visible to Roslyn analyzers that naively examine a type's attributes. One option is to have Roslyn special case this attribute: when the compiler sees the metadata bit, it would act as if the `Serializable` attribute had been specified.